feature-libs/quote/components/quote-seller-edit/quote-seller-edit.component.service.spec.ts

```diff
@@ -0,0 +1,241 @@
+      expect(dateWithTime).toContain('T');
+      expect(dateWithTime).toContain(TimeUtils.getLocalTimezoneOffset());
+    });
+  });
+
+  describe('removeTimeFromDate', () => {
+    it('should remove the time part from a time stamp', () => {
+      expect(service.removeTimeFromDate(EXPIRATION_TIME_AS_STRING)).toBe(
+        EXPIRATION_DATE_AS_STRING
+      );
+    });
+
+    it('should do nothing for undefined time stamp', () => {
+      expect(service.removeTimeFromDate(undefined)).toBeUndefined();
+    });
+  });
+
+  describe('performValidationAccordingToMetaData', () => {
+    it('should accept input using group and decimal separators', () => {
+      expect(
+        service['performValidationAccordingToMetaData'](
+          '1.000,76',
+          '.',
+          ',',
+          10
+        )
+      ).toBe(false);
+    });
+    it('should not accept input with 2 decimal separators', () => {
+      expect(
+        service['performValidationAccordingToMetaData'](
+          '1,000,76',
+          '.',
+          ',',
+          10
+        )
+      ).toBe(true);
+    });
+  });
+
+  describe('getMaximumNumberOfTotalPlaces', () => {
+    it('should compile number of total places from total if no quote discount is present, taking 2 decimal points into account ', () => {
+      expect(service.getMaximumNumberOfTotalPlaces(quote)).toBe(
+        Math.log10(TOTAL_PRICE) + 3
+      );
+    });
+
+    it('should compile number of total places from absolute discount if that exceeds total ', () => {
+      quote.quoteDiscounts = { value: DISCOUNT_RATE };
+      expect(service.getMaximumNumberOfTotalPlaces(quote)).toBe(
+        Math.log10(DISCOUNT_RATE) + 3
+      );
+    });
+
+    it('should fall back to price value 1 if no values are available at all (will not happen in production) ', () => {
+      quote.totalPrice.value = undefined;
+      expect(service.getMaximumNumberOfTotalPlaces(quote)).toBe(3);
+    });
+
+    it('should round properly according to log10', () => {
+      const quote2000: Quote = { ...quote, totalPrice: { value: 2000 } };
+      const quote3000: Quote = { ...quote, totalPrice: { value: 3000 } };
+      expect(service.getMaximumNumberOfTotalPlaces(quote2000)).toBe(
+        service.getMaximumNumberOfTotalPlaces(quote3000)
+      );
+      expect(service.getMaximumNumberOfTotalPlaces(quote)).toBe(
+        service.getMaximumNumberOfTotalPlaces(quote3000)
+      );
+    });
+
+    it('should round properly according to log10 for different maximum totals', () => {
+      const quote999: Quote = { ...quote, totalPrice: { value: 999 } };
+      const quote100: Quote = { ...quote, totalPrice: { value: 100 } };
+      expect(service.getMaximumNumberOfTotalPlaces(quote999)).toBe(
+        service.getMaximumNumberOfTotalPlaces(quote100)
+      );
+      expect(service.getMaximumNumberOfTotalPlaces(quote)).toBe(
+        service.getMaximumNumberOfTotalPlaces(quote999) + 1
+      );
```

Comment:
those both test seem to assume that Math.log10 is used to count the number of places which is an implementation detail.  in the end they just check
2000 => 6
3000 => 6
999 => 5
1000 => 6

Reply:
Partially agree: I wasn't aware of Math.floor and wanted to test my rounding => too many tests if one believes that Math.floor does what it's supposed to do.
But let's still keep the second test, otherwise the rounding is not covered at all

Reply:
Renamed the test to clarify what it's supposed to do (not referring to the implementation details)